### PR TITLE
Feat: Extract and adds tokens to Dropdown

### PR DESCRIPTION
**Pull request by hellofanny**
Ported from https://github.com/vtex-sites/nextjs.store/pull/118

## What's the purpose of this pull request?

- Extracts `Dropdown` component from `Breadcrumb`
- Applies the new https://github.com/vtex-sites/base.store/pull/407 to `Dropdown`: new scoped tokens linked to global ones.
- Adds `size` prop to `DropdownMenu`

## How does it work?
Some visual changes were made to the Dropdown on `Breadcrumb`.
|[Before](https://gatsby-store-storybook-faststore.vercel.app/?path=/docs/molecules-breadcrumb--default)|[Now](https://gatsby-store-storybook-git-feat-theming-dropdo-c8712b-faststore.vercel.app/?path=/docs/molecules-breadcrumb--default)|
|-|-|
before|now

- Icon default color is blue;
- Added missing light gray divider between items;

Tokens applied only for `DropdownMenu` and `DropdownItem`, `DropdownButton` will be revisit soon and doesn't have a default style.

## How to test it?

- The dropdown should be working fine on the `Breadcrumb` as before these changes, but with the previously mentioned visual changes.
- Check the `Dropdown` documentation in the [preview](https://gatsby-store-storybook-git-feat-theming-dropdo-c8712b-faststore.vercel.app/?path=/docs/molecules-dropdown-default--default-story) 

## Checklist

<em>You may erase this after checking them all ;)</em>

**Changelog**
- [x] Added an entry in the `CHANGELOG.md` at the beginning of its due section. [The latest version should comes first](https://keepachangelog.com/en/1.0.0/#:~:text=The%20latest%20version%20comes%20first.).
- [x] Added the PR number with the PR link at the entry in the `CHANGELOG.md`. E.g., *New items in the `pull_request_template.md` ([#12](https://github.com/vtex-sites/gatsby.store/pull/12))*

**PR Description**
- [x] Added a label according to the PR goal - `Breaking change`, `Enhancement`, `Bug` or `Chore`.
- [x] Added the component, hook, or pathname in-between backticks (``) *- If applicable*. E.g., *`ComponentName` component*.
- [x] Identified the function or parameter in the PR *- If applicable*. E.g., *`useWindowDimensions` hook*.

**Documentation**
- [x] PR description
- [x] Added to/Updated the Storybook - *if applicable*.
- [ ] For documentation changes, ping @ carolinamenezes, @ PedroAntunesCosta or @ Mariana-Caetano to review and update.

### CODE REVIEW

File: src/components/ui/Breadcrumb/breadcrumb.scss [hunk 1]
```diff
@@ -78,7 +78,7 @@
       }
     }
 
-    [data-store-dropdown-button] {
+    [data-fs-dropdown-button] {
```
Comment (filipewl) [inline src/components/ui/Breadcrumb/breadcrumb.scss:81]: Looks like we didn't do this for NextJS, so maybe it's something we should not forget to add there later.

https://github.com/vtex-sites/nextjs.store/blob/2460a62cdae585628aee95b3b92540fd0106b54a/src/components/ui/Breadcrumb/breadcrumb.scss#L81

**Review by filipewl — APPROVED** [review]
Looks good to me once you take a look at the comments I left! I found some minor inconsistencies from the NextJS version that I think we can keep the same.

**Review by eduardoformiga — APPROVED** [review]
LGTM!
It's worth taking a look at @filipewl comments 😉

File: src/components/ui/Breadcrumb/breadcrumb.scss (hunk 1 above)
Comment (hellofanny) [reply src/components/ui/Breadcrumb/breadcrumb.scss:81]: I've noticed that I missed some details on NextJS version while doing this one 😓 so I've draft a PR to sync them https://github.com/vtex-sites/nextjs.store/pull/123

Thanks for the heads up :)

File: src/components/ui/Breadcrumb/Breadcrumb.tsx [hunk 2]
```diff
@@ -62,23 +61,23 @@ function BaseBreadcrumb({
...
+          <DropdownMenu>
             {mediumItems.map(({ item, name }, index) => (
-              <UIDropdownItem
-                data-store-dropdown-menu-item
+              <DropdownItem
+                data-breadcrumb-dropdown-item
                 onClick={() => navigate(item)}
                 key={String(index)}
               >
                 <Icon name="ArrowElbowDownRight" width={24} height={24} />
```
[the remark this answers is not in the document]
Comment (hellofanny) [reply src/components/ui/Breadcrumb/Breadcrumb.tsx:75]: oops! I'll fix that.

**Pull request merged**